File: iosNativeApp/iosNativeApp/Tools/SettingsController.swift
import Foundation

final class SettingsController {
    private static let key = "currencies"

    static func saveCurrencies(_ currencies: Set<String>) {
        UserDefaults.standard.set(currencies.joined(separator: ","), forKey: key)
    }

    static func getCurrencies() -> Set<String> {
        guard let raw = UserDefaults.standard.string(forKey: key), !raw.isEmpty else {
            return []
        }
        return Set(raw.split(separator: ",").map(String.init))
    }
}

File: sharedUI/src/commonMain/kotlin/io/github/currencyconverter/tools/SettingsController.kt
package io.github.currencyconverter.tools

expect object SettingsController {

    fun saveCurrencies(currencies: Set<String>)
    fun getCurrencies(): Set<String>
}

internal fun String?.toSet(): Set<String>? =
    if (this?.isNotBlank() == true) {
        this.split(",")
            .toSet()
    } else {
        null
    }